internal/actions/setvar.go

```diff
@@ -150,11 +150,17 @@ func (a *setvarFn) evaluateTxCollection(r plugintypes.RuleMetadata, tx plugintyp
 		if len(value) > 1 {
 			val, err = strconv.Atoi(value[1:])
 			if err != nil {
-				tx.DebugLogger().Error().
-					Str("var_value", value).
-					Int("rule_id", r.ID()).
-					Err(err).
-					Msg("Invalid value")
+				// If the variable doesn't exist, we would need to raise an error. Otherwise, it should be the same value.
+				if strings.HasPrefix(value[1:], "tx.") {
```

Comment:
shouldn't this be `%{tx.`? instead of `tx.`

Reply:
It is either resolve to the value or stays as tx.missingValue.

Reply:
But according to https://github.com/corazawaf/coraza/pull/1125/files#diff-488435c221aebedf4137cb98424234aab81decfd749827ab96e188c53bd53c9fR90-R92 it returns a syntax error.

Reply:
Here's the value: <img width="987" alt="Screenshot 2024-08-05 at 2 38 24 PM" src="https://github.com/user-attachments/assets/8523e85c-a5b1-426a-b7a8-8eb8bb16c729">

Reply:
I see. It is still not clear to me how we should deal with the interpolation error in this code path. Let me check it.

Reply:
Sure, thanks!

Reply:
Hey @jcchavezs , 
Any update on how this error could be handled? 

Thanks!

Reply:
Ok I found the issue, when we compile a macro e.g. `%{tx.var}`, the text for the macro becomes `tx.var` and not `%{tx.var}` and when the key does not exist we return the `text` which in this case is `tx.var`. I will discuss that over another PR.